### PR TITLE
fix(ToggleSwitch): Don't force end position when ToggleSwitch is unloaded

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/unoplatform/uno/issues/12643

I had a previous fix for this, but it's either that some other changes have competed with #12643 or that my PR fixed the issue in a certain situation but not all scenarios. I can't recall exactly the code sample I was testing my PR against. Anyway, this is a small fix to ensure we don't set the translation incorrectly when unloaded.

Setting it incorrectly when unloaded will later affect the values set for the minimum and maximum knob translation.

I'm not sure how to go about testing this.

## PR Type

What kind of change does this PR introduce?

### <samp>🤖 Generated by Copilot at 5ad2ca2</samp>

Fixed potential errors in the `ToggleSwitch` control by adding validation and initialization logic. Modified `ToggleSwitch.mux.cs` to prevent invalid knob positions and check for null values.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs [hunk 1]
```diff
@@ -107,10 +108,15 @@ private protected override void ChangeVisualState(bool useTransitions)
...
+
 				// TODO Uno specific: We must force the knob to position, because
 				// we don't yet support the transitions which do this automatically
 				// in XAML template.
-				ForceSwitchKnobEndPosition();
+				if (IsLoaded)
+				{
+					Debug.Assert(GetMaxOffset() > 0);
+					ForceSwitchKnobEndPosition();
+				}
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs:119]: Are we able to validate this somehow by inspecting the children positions?

Comment (agneszitte) [reply src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs:119]: Can we also add tests for this @Youssef1313 / @jeromelaban ?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/ToggleSwitch/ToggleSwitch.mux.cs:119]: I'll try with getting `_minKnobTranslation` and `_maxKnobTranslation` via reflection and assert against these. This is the core thing that breaks ToggleSwitch. The intent of this change isn't only about the "initial" position of the ToggleSwitch, but it's also when dragging, we get negative _minKnobTranslation and then dragging is only allowed in the reverse direction (outside the bounds of the control)

**Review by jeromelaban — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

**Comment by agneszitte** [comment]
@Mergify backport legacy/4x

**Pull request merged**